src/test/scala/com/github/timgent/dataflare/metrics/MetricCalculatorTest.scala

```diff
@@ -181,4 +181,56 @@ class MetricCalculatorTest extends AnyWordSpec with DatasetSuiteBase with Matche
       )
     }
   }
+
+  "MinValueMetricCalculator" should {
+    lazy val ds = Seq(
+      NumberString(1, "a"),
+      NumberString(2, "b"),
+      NumberString(3, "c"),
+      NumberString(3, "d"),
+      NumberString(3, "d") // not distinct
+    ).toDS
+
+    "calculate the min of values for a given column" in {
+      testMetricAggFunction(
+        ds,
+        MinValueMetricCalculator[LongMetric]("number", MetricFilter.noFilter),
+        LongMetric(1)
+      )
+    }
+
+    "apply the provided filter before calculating the compliance in a DataFrame" in {
+      testMetricAggFunction(
+        ds,
+        MinValueMetricCalculator[LongMetric]("number", MetricFilter(col("str") =!= lit("d"), "not d")),
+        LongMetric(Long.MinValue)
+      )
+    }
+  }
+
+  "MaxValueMetricCalculator" should {
+    lazy val ds = Seq(
+      NumberString(1, "a"),
+      NumberString(2, "b"),
+      NumberString(3, "c"),
+      NumberString(3, "d"),
+      NumberString(3, "d") // not distinct
+    ).toDS
+
+    "calculate the min of values for a given column" in {
+      testMetricAggFunction(
+        ds,
+        MaxValueMetricCalculator[LongMetric]("number", MetricFilter.noFilter),
+        LongMetric(3)
+      )
+    }
+
+    "apply the provided filter before calculating the compliance in a DataFrame" in {
+      testMetricAggFunction(
+        ds,
+        MaxValueMetricCalculator[LongMetric]("number", MetricFilter(col("str") =!= lit("d"), "not d")),
```

Comment:
We should test with a filter that would change the output expected metric to make it a better test

Without the filter you would get max of 3. With the filter you gave you would also get max of 3. So I think the test could be improved

Reply:
Have changed the ds values to show different results with and without filters